Cache the `oxide` targets

RobinMalfait opened the pull request:
This will cache the oxide related build files to hopefully minimize the
amount of Rust compiling.

<!--

👋 Hey, thanks for your interest in contributing to Tailwind!

**Please ask first before starting work on any significant new features.**

It's never a fun experience to have your pull request declined after investing a lot of time and effort into a new feature. To avoid this from happening, we request that contributors create an issue to first discuss any significant new features. This includes things like adding new utilities, creating new at-rules, or adding new component examples to the documentation.

https://github.com/tailwindcss/tailwindcss/blob/master/.github/CONTRIBUTING.md

-->

RobinMalfait committed:
cache the `oxide` targets

This will cache the oxide related build files to hopefully minimize the
amount of Rust compiling.

RobinMalfait committed:
tmp: drop turbo

thecrypticace commented on .github/workflows/ci.yml:
```diff
@@ -46,7 +46,19 @@ jobs:
+        with:
+          path: |
+            ./oxide/crates/target
+            ./oxide/crates/node/target
```
Aren't these two paths wrong? Afaik it's `./oxide/target`

RobinMalfait replied:
Yep totally, was just playing with things and saw that I had a `target` folder in `./oxide/crates/node` so added them but that was wrong.

Reason it is wrong is because I use this on my personal machine:
```sh
export CARGO_TARGET_DIR=~/cargo-target
```
Which basically uses `~/cargo-target` as a shared folder for _all_ my Rust programs. Kinda nice because I can do this:
```sh
PATH=~/cargo-target/release:$PATH
```
And all my binaries are available from everywhere.

I think for now we just want `./oxide/target` indeed, but the moment we have more Rust binaries, this custom target might be nice to introduce.

RobinMalfait committed:
cache `./oxide/target/`

Thanks @thecrypticace!

RobinMalfait committed:
no need to cache `oxide` files

This will already be cached by GitHub actions. This should save us many
GBs on Vercel.com and Rust (or Cargo) is way better in using existing
cached information so this mix of caches between Turbo and GitHub
actions is kind of nice.

RobinMalfait committed:
Revert "tmp: drop turbo"

This reverts commit 22761d3a6669205a65695c6887bc9fda5f495adf.

RobinMalfait committed:
improve caching for integration tests and insiders release
